pkg/controller/storagecluster/reconcile.go

```diff
@@ -268,7 +219,14 @@ func (r *ReconcileStorageCluster) Reconcile(request reconcile.Request) (reconcil
 		r.ensureCephConfig,
 		r.ensureCephCluster,
 		r.ensureNoobaaSystem,
-	} {
+	}
+	// for external cluster, we have a different set of ensure functions
+	if instance.Spec.ExternalStorage.Enable {
```

Comment:
> Why define it and then override, when we can define once as needed.

This is generally the style/approach to avoid an unnecessary if/else. We could have added an if/else (as you suggested) if we were calling the functions, but here we are merely assigning the function literals (not the return values after calling them).
An if/else may usually clutters the code and even that case we are **not** avoiding any overriding (it will look something like this),
```go
var ensureFs []ensureFunc // here it is by-default initialized to nil
if !instance.Spec.ExternalStorage.Enable {
      ensureFs = []ensureFunc{  // we are re-defining either here or
            // Add support for additional resources here
            r.ensureStorageClasses,
            r.ensureCephObjectStores,
            r.ensureCephObjectStoreUsers,
            r.ensureCephBlockPools,
            r.ensureCephFilesystems,

            r.ensureCephConfig,
            r.ensureCephCluster,
            r.ensureNoobaaSystem,
      }
} else {
      // for external cluster, we have a different set of ensure functions
      ensureFs = []ensureFunc{  // re-defining here
            r.ensureCephCluster,
      }
}
```

I have no reservations. So if you feel that code readability is better by having if/else way, will change it accordingly...

Reply:
As most of us feel if/else preferred way to go, made the changes... Please take a look